java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -0,0 +1,210 @@
+        // process block exit, which is unconditional jump such as goto-statement or return-statement
+        handleBlockExit(programPosition);
+      }
+    }
+
+    handleEndOfExecutionPath();
+    executeCheckEndOfExecution();
+    // Cleanup:
+    workList = null;
+    node = null;
+    programState = null;
+    constraintManager = null;
+  }
+
+  private void executeInstruction(BytecodeCFGBuilder.Instruction instruction) {
+    if(!checkerDispatcher.executeCheckPreStatement(instruction)) {
+      return;
+    }
+    switch (instruction.opcode) {
+      case Opcodes.ARETURN:
+        programState.storeExitValue();
+        break;
+      case Opcodes.ACONST_NULL:
+        programState = programState.stackValue(SymbolicValue.NULL_LITERAL);
+        break;
+      case Opcodes.ALOAD:
+      case Opcodes.DLOAD:
+      case Opcodes.FLOAD:
+      case Opcodes.ILOAD:
+        break;
+      case Opcodes.LLOAD:
+      case Opcodes.AALOAD:
+        break;
+      case Opcodes.BALOAD:
+      case Opcodes.CALOAD:
+      case Opcodes.DALOAD:
+      case Opcodes.FALOAD:
+      case Opcodes.IALOAD:
+      case Opcodes.LALOAD:
+      case Opcodes.SALOAD:
+        break;
+      case Opcodes.LDC:
+        SymbolicValue symbolicValue = constraintManager.createSymbolicValue(instruction);
+        programState = programState.stackValue(symbolicValue);
+        programState = programState.addConstraint(symbolicValue, ObjectConstraint.NOT_NULL);
+        break;
+      default:
+        // do nothing
+    }
+    checkerDispatcher.executeCheckPostStatement(instruction);
+  }
+
+  private void handleBlockExit(ProgramPoint programPosition) {
+    programPosition.block.successors().forEach(b -> enqueue(new ProgramPoint(b), programState));
+  }
+
+  private void executeCheckEndOfExecution() {
+    // TODO callback to checks at end of execution
+  }
+
+  private Iterable<ProgramState> startingStates(Symbol.MethodSymbol symbol, ProgramState currentState) {
+    // TODO : deal with parameter annotations, equals methods etc.
+    int arity = symbol.parameterTypes().size();
+    ProgramState state = currentState;
+    if(!symbol.isStatic()) {
+      // Add a sv for "this"
+      SymbolicValue thisSV = constraintManager.createSymbolicValue((BytecodeCFGBuilder.Instruction) null);
+      methodBehavior.addParameter(thisSV);
+      state = currentState.addConstraint(thisSV, ObjectConstraint.NOT_NULL);
+    }
+    IntStream.range(0, arity).forEach(i -> methodBehavior.addParameter(constraintManager.createSymbolicValue((BytecodeCFGBuilder.Instruction) null)));
+    return Collections.singletonList(state);
+  }
+
+  private void setNode(ExplodedGraph.Node node) {
+    this.node = node;
+    programPosition = this.node.programPoint;
+    programState = this.node.programState;
+  }
+
+  void enqueue(ProgramPoint pp, ProgramState programState) {
+    int nbOfExecution = programState.numberOfTimeVisited(pp);
+    if (nbOfExecution > MAX_EXEC_PROGRAM_POINT) {
+        return;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this line start at column 7. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)